x-pack/plugin/snapshot-repo-test-kit/src/main/java/org/elasticsearch/repositories/blobstore/testkit/analyze/RepositoryAnalyzeAction.java

```diff
@@ -521,6 +521,13 @@ public void run() {
                 final long targetLength = blobSizes.get(i);
                 final boolean smallBlob = targetLength <= MAX_ATOMIC_WRITE_SIZE; // avoid the atomic API for larger blobs
                 final boolean abortWrite = smallBlob && request.isAbortWritePermitted() && rarely(random);
+                final boolean doCopy = random.nextBoolean();
```

Comment:
I think I'd be inclined to have the `RepositoryAnalyzeAction` control the name of the copied blob rather than just setting a flag. That way we can record it in `AsyncAction#expectedBlobs` and check it exists in the listing at the end.

Also this can be a rare action, we don't need to copy half of the blobs (+50% storage size on average).

Reply:
Actually on reflection it's probably simplest not to put this in `expectedBlobs` (for now anyway). Not only do we have to think about the `doEarlyCopy` case but also we need to handle the fact that GCS/Azure will fail the copy with a UOE.

I'd still be inclined to have the `RepositoryAnalyzeAction` specify the blob name rather than just set a flag in the request because we'll want to do these things in future, but that's the only change I'd suggest in this areal.

Reply:
I pushed d9db70da06563a10d6753843445c86c279a87a31 which should fix the `FailureIT`